nushell_completions.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2013-2022 The Cobra Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cobra
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+)
+
+func (c *Command) GenNushellCompletion(w io.Writer, includeDesc bool) error {
+	buf := new(bytes.Buffer)
+	WriteStringAndCheck(buf, fmt.Sprintf(`
+# An external completer that works with any cobra based
```

Comment:
Please replace all tabs with four spaces in the shell script.

Reply:
Will do.. I had them as tabs before as that was the LSP default and the go style guide says tabs over spaces.. I like spaces better myself.

Reply:
done.